main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -910,14 +909,54 @@ object ParsedAst {
     case class Cast(exp: ParsedAst.Expression, tpe: Option[ParsedAst.Type], pur: Option[ParsedAst.Type], sp2: SourcePosition) extends ParsedAst.Expression
 
     /**
-      * Try Catch Expression.
+      * Do Expression.
       *
-      * @param sp1   the position of the first character in the expression.
-      * @param exp   the guarded expression.
-      * @param rules the catch rules.
-      * @param sp2   the position of the last character in the expression.
+      * @param sp1  the position of the first character in the expression.
+      * @param op   the effect operation.
+      * @param args the arguments to the operation.
+      * @param sp2  the position of the last character in the expression.
+      */
+    case class Do(sp1: SourcePosition, op: Name.QName, args: Seq[ParsedAst.Argument], sp2: SourcePosition) extends Expression
+
+    /**
+      * Resume Expression.
+      *
+      * @param sp1  the position of the first character in the expression.
+      * @param args the arguments to the continuation.
+      * @param sp2  the position of the last character in the expression.
       */
-    case class TryCatch(sp1: SourcePosition, exp: ParsedAst.Expression, rules: Seq[ParsedAst.CatchRule], sp2: SourcePosition) extends ParsedAst.Expression
+    case class Resume(sp1: SourcePosition, args: Seq[ParsedAst.Argument], sp2: SourcePosition) extends Expression
+
+    /**
+      * Try Expression.
+      *
+      * @param sp1            the position of the first character in the expression.
+      * @param exp            the guarded expression.
+      * @param catchOrHandler the handler (catch/with) of the try expression.
+      * @param sp2            the position of the last character in the expression.
+      */
+    case class Try(sp1: SourcePosition, exp: ParsedAst.Expression, catchOrHandler: CatchOrHandler, sp2: SourcePosition) extends ParsedAst.Expression
+
+    /**
+      * An enum representing the handler of a `try` expression.
+      */
+    sealed trait CatchOrHandler
```

Comment:
Usually this helper types are plased later in the file right? Not in here in Expression?